internal/uri.go

```diff
@@ -0,0 +1,45 @@
+package internal
+
+import (
+	"net/url"
+	"os"
+	"strings"
+)
+
+// https://buildpacks.io/docs/reference/config/package-config/#dependencies-list-optional
+func IsArchive(dependency PackageConfigDependency) bool {
+	if IsCnbRegistry(dependency) {
+		return false
+	}
+
+	if IsDocker(dependency) {
+		return false
+	}
+
+	return IsCnbFile(dependency)
+}
+
+func IsCnbFile(dependency PackageConfigDependency) bool {
+	if uri, err := url.Parse(dependency.URI); err != nil {
```

Comment:
I think is better to do something like what you do on the isDirectory. To look if the file exists and is an archive